runtime/cudaq/cudaq.cpp

```diff
@@ -280,7 +280,16 @@ void unset_noise() {
   platform.set_noise(nullptr);
 }
 
-void set_random_seed(std::size_t seed) { nvqir::setRandomSeed(seed); }
+static std::size_t cudaq_random_seed = 0;
```

Comment:
Thinking out loud, open to other suggestions - instead of introducing a global static variable (and then subsequently worrying about multiple threads changing it), perhaps it would be better to just keep the seed where it is on `CircuitSimulator` (which is `thread_local`) and provide a `nvqir::getRandomSeed()` function. Then in `broadcast.h`, you can get the one seed set for the current calling thread and share it amongst all the threads in the broadcast functor.

Reply:
The seed isn't technically stored anywhere in `CircuitSimulator`. Only the `RandomDevices` object is, and that doesn't expose originating seed. However, if I add `thread_local` to this new `cudaq_random_seed` variable, I believe that should give you the desired effect, and I think that would be better than implementing that logic in both `QppCircuitSimulator` and `CuStateVecCircuitSimulator`. I'll give that a shot and post an update soon.

Reply:
Updated in [830aeea](https://github.com/NVIDIA/cuda-quantum/pull/648/commits/830aeeae5066269a8bc8a0b07fb8d18aa06c588a). Letme know what you think.